diff -Naupr gst-fluendo-asfdemux-0.10.76/src/fluasfdemux.h gst-fluendo-asfdemux-0.10.76-copy/src/fluasfdemux.h
--- gst-fluendo-asfdemux-0.10.76/src/fluasfdemux.h	2014-10-21 15:53:46.000000000 +0200
+++ gst-fluendo-asfdemux-0.10.76-copy/src/fluasfdemux.h	2014-11-20 19:35:13.421324901 +0100
@@ -18,6 +18,7 @@
 #endif
 #if BUILD_DX
 #include "dxdrm/DxDrmClient.h"
+#define DEBUG_DRM 1
 #endif
 
 #if GST_CHECK_VERSION (1,0,0)
